heliopy/data/mms.py

```diff
@@ -7,131 +7,1773 @@
+            except:
+                os.chdir(pwd)
+                raise
+
+            for file in glob.glob(os.path.basename(path)):
+                result.append(os.path.join(root, file))
+
+        os.chdir(pwd)
+
+        return result
+
+    def login(self, username=None, password=None):
+        '''
+        Log-In to the SDC
+
+        Parameters
+        ----------
+        username (str):     Account username
+        password (str):     Account password
+        '''
+
+        # Ask for inputs
+        if username is None:
+            username = input('username: ')
+        if password is None:
+            password = input('password: ')
+
+        # Save credentials
+        self._session.auth = (username, password)
+
+    def name2path(self, filename):
+        '''
+        Convert remote file names to local file name.
+
+        Directories of a remote file name are separated by the '/' character,
+        as in a web address.
+
+        Parameters
+        ----------
+        filename : str
+            File name for which the local path is desired.
+
+        Returns
+        -------
+        path : str
+            Equivalent local file name. This is the location to
+            which local files are downloaded.
+        '''
+        parts = filename.split('_')
+
+        # burst data selection directories and file names are structured as
+        #   - dirname:  sitl/[type]_selections/
+        #   - basename: [type]_selections_[optdesc]_YYYY-MM-DD-hh-mm-ss.sav
+        # To get year, index from end to skip optional descriptor
+        if parts[1] == 'selections':
+            path = os.path.join(self.data_root, 'sitl',
+                                '_'.join(parts[0:2]),
+                                filename)
+
+        # Burst directories and file names are structured as:
+        #   - dirname:  sc/instr/mode/level[/optdesc]/YYYY/MM/DD/
+        #   - basename: sc_instr_mode_level[_optdesc]_YYYYMMDDhhmmss_vX.Y.Z.cdf
+        # Index from end to catch the optional descriptor, if it exists
+        elif parts[2] == 'brst':
+            path = os.path.join(self.data_root, *parts[0:-2],
+                                parts[-2][0:4], parts[-2][4:6],
+                                parts[-2][6:8], filename)
+
+        # Survey (slow,fast,srvy) directories and file names are structured as:
+        #   - dirname:  sc/instr/mode/level[/optdesc]/YYYY/MM/
+        #   - basename: sc_instr_mode_level[_optdesc]_YYYYMMDD_vX.Y.Z.cdf
+        # Index from end to catch the optional descriptor, if it exists
+        else:
+            path = os.path.join(self.data_root, *parts[0:-2],
+                                parts[-2][0:4], parts[-2][4:6], filename)
+
+        return path
+
+    def parse_file_name(self, filename):
+        '''
+        Parse an official MMS file name. MMS file names are formatted as
+            sc_instr_mode_level[_optdesc]_tstart_vX.Y.Z.cdf
+        where
+            sc:       spacecraft id
```

Comment:
I think for these to be indented they need to be part of a list, and have `- ` added before each item.